### PR TITLE
Fix granular `jwt` and `saml` flags on FE

### PR DESCRIPTION
**Pull request by nemanjaglumac**
#32399 introduced more granular feature flags for `sso`, but didn't update the frontend to reflect this.
That resulted in E2E test failures and in the absence of `jwt` and `saml` options from the admin authentication options when the token is present.

### How to test?
Both these specs should now pass in CI and/or locally:
- `e2e/test/scenarios/admin/settings/sso/jwt.cy.spec.js`
- `e2e/test/scenarios/admin/settings/sso/saml.cy.spec.js`

You can also check this manually by going to `/admin/settings/authentication` **with the active PRO token**. Make sure you can see JWT and SAML cards.

**Before**
image

**After**
image

### CODE REVIEW

File: enterprise/frontend/src/metabase-enterprise/auth/index.js [hunk 1]
```diff
@@ -29,14 +29,14 @@ PLUGIN_ADMIN_SETTINGS_UPDATES.push(sections =>
       description: null,
       noHeader: true,
       widget: SamlAuthCard,
-      getHidden: () => !hasPremiumFeature("sso"),
+      getHidden: () => !hasPremiumFeature("sso_saml"),
```
Comment (nemanjaglumac) [inline enterprise/frontend/src/metabase-enterprise/auth/index.js:32]: I think that #32399 completely removed `sso` flag.
Not sure if we need it as the fallback for downgrades? cc @NevRA 

Also, please note that all other feature flags are kebab-cased but these specific flags are set as the snake_case in #32399.

tl;dr
Not sure if this particular fix is even needed on the FE, or do we want to being back `sso` and to adjust granular flags to be kebab-cased.

cc @ranquild

**Comment by nemanjaglumac** [comment]
Closing because https://github.com/metabase/metabase/pull/32317 already solves this and much more, but I didn't see it.
